### PR TITLE
ANN: Get rid of `const_generic` feature, support `adt_const_params` feature

### PR DESCRIPTION
**Pull request by mchernyavsky**
Relates to https://github.com/intellij-rust/intellij-rust/issues/3985#issuecomment-913837227.

changelog: Get rid of `const_generic` feature, support `adt_const_params` feature

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt [hunk 1]
```diff
@@ -1512,9 +1506,10 @@ fun RsDiagnostic.addToHolder(holder: RsProblemsHolder) {
     holder.registerProblem(descriptor)
 }
 
-private val PreparedAnnotation.fullDescription: String get() {
-    return "<html>${htmlHeader(errorCode, escapeString(header))}<br>${escapeString(description)}</html>"
-}
+private val PreparedAnnotation.fullDescription: String
+    get() {
+        return "<html>${htmlHeader(errorCode, escapeString(header))}<br>${escapeString(description)}</html>"
+    }
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt:1512]:
```suggestion
    get() = "<html>${htmlHeader(errorCode, escapeString(header))}<br>${escapeString(description)}</html>"
```

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 2]
```diff
@@ -1341,38 +1341,14 @@ private fun checkDuplicates(
...
-                MIN_CONST_GENERICS
-            } else {
-                CONST_GENERICS
-            }
-        }
-        else -> CONST_GENERICS
+    val typeReference = constParameter.typeReference
+    val ty = typeReference?.type
+    if (typeReference != null && ty !is TyInteger && ty !is TyBool && ty !is TyChar) {
+        ADT_CONST_PARAMS.check(holder, typeReference, "adt const params")
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt:1350]: Can we also check [`E0741`](https://doc.rust-lang.org/error-index.html#E0741) here (maybe in separate PR)?

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: https://github.com/intellij-rust/intellij-rust/pull/7897

**Pull request merged**